envoyextensions/extensioncommon/basic_envoy_extender.go

```diff
@@ -168,13 +201,41 @@ func (b *BasicEnvoyExtender) patchClusters(config *RuntimeConfig, clusters Clust
 		if err != nil {
 			resultErr = multierror.Append(resultErr, fmt.Errorf("error patching cluster %q: %w", nameOrSNI, err))
 		}
+		if !patched {
+			patchedCluster = cluster
+		}
+
+		// We patch cluster load assignments directly above for EDS, but also here for CDS,
+		// since updates can come from either.
+		if patchedCluster.LoadAssignment != nil {
+			patchedClusterLoadAssignment, patched, err := b.Extension.PatchClusterLoadAssignment(config, patchedCluster.LoadAssignment)
+			if err != nil {
+				resultErr = multierror.Append(resultErr, fmt.Errorf("error patching load assignment for cluster %q: %w", nameOrSNI, err))
+			} else if patched {
+				patchedCluster.LoadAssignment = patchedClusterLoadAssignment
+			}
+		}
+
+		patchedClusters[nameOrSNI] = patchedCluster
+	}
+	return patchedClusters, resultErr
+}
+
+func (b *BasicEnvoyExtender) patchClusterLoadAssignments(config *RuntimeConfig, clusterLoadAssignments ClusterLoadAssignmentMap) (ClusterLoadAssignmentMap, error) {
+	var resultErr error
+
+	for nameOrSNI, clusterLoadAssignment := range clusterLoadAssignments {
+		patchedClusterLoadAssignment, patched, err := b.Extension.PatchClusterLoadAssignment(config, clusterLoadAssignment)
+		if err != nil {
+			resultErr = multierror.Append(resultErr, fmt.Errorf("error patching cluster load assignment %q: %w", nameOrSNI, err))
+		}
 		if patched {
-			patchedClusters[nameOrSNI] = patchedCluster
+			clusterLoadAssignments[nameOrSNI] = patchedClusterLoadAssignment
 		} else {
-			patchedClusters[nameOrSNI] = cluster
+			clusterLoadAssignments[nameOrSNI] = clusterLoadAssignment
 		}
 	}
-	return patchedClusters, resultErr
+	return clusterLoadAssignments, resultErr
```

Comment:
This diff is odd-looking due to similar-shaped code being introduced below code that was edited; the red here is the modified end of `patchClusters` above, and the green is the tail end of the new `patchClusterLoadAssignments`.